### PR TITLE
Removing MKLSolversApplication

### LINKED ISSUE #1316: MKLSolversApplication
I wanted to open a discussion about the MKLSolverApplication. As far as I know, it is currently not working. Since the MKLPardiso solvers are included in the EigenSolversApplication now, I think we might discuss what should be done with the MKLSolversApplication.

### PR DESCRIPTION
**Pull request by philbucher**
This PR removes the outdated MKLSolversApplication. This was discussed in #1316 

@roigcarlo I removed the mkl stuff from the compilation scripts, can you please have a look to make sure it doesn't break anything?

@RiccardoRossi I also removed the cmake find packages, I hope this is ok

I tried to remove everything, but there are still many occurences if one searches kratos for "mkl"
I am quite sure these are outdated but I still want to mention it, so that the developers can remove / update the outdated stuff. Among them are:
- ConstritutiveLawsApplication @josep-m-carbonell 
- DEM @maceligueta 
- FluidDynamics and IncompressibleFluid @jcotela 
- Freezing Soil
- MeshingApplication @loumalouomega 
- MixedElementApp
- OpenCLApplication
- MetisApplication @roigcarlo 
- PfemSolidMechanics @josep-m-carbonell 
- structural_application @RiccardoRossi 
- swimming_DEM_application @maceligueta 
- ThermoMechanicalApplication 

Btw some mkl solvers are now available through the EigenSolversApplication, see the readme in the application

closes #1316

### CODE REVIEW

**Comment by jcotela** [comment]
Hi @philbucher regarding the FluidDynamicsApplication, incompressible_fluid_application and ThermoMechanicalApplication, the references to MKL are safe to remove. Should I do it on this PR?

**Comment by philbucher** [comment]
@jcotela please feel free to push to this branch

**Comment by jcotela** [comment]
@philbucher I've just removed all occurences of MKL in files I "manage". I think in most cases I could have deleted the files outright, but I can do the cleanup later.

I've also removed those from Metis, @roigcarlo and I believe they were there as a copy-paste error anyway.

**Comment by philbucher** [comment]
The following applications are left to clean:
- DEM @maceligueta 
- Freezing Soil
- MixedElementApp
- OpenCLApplication
- structural_application @RiccardoRossi 
- swimming_DEM_application @maceligueta

**Comment by josep-m-carbonell** [comment]
ConstritutiveLawsApplication is not under my responsibility and it is deprecated
It can be replaced by UmatApplication
For the PfemSolidMechanics, you can go ahead, mkl is not used.

**Comment by loumalouomega** [comment]
I work in the meshing application, but I am not the responsible, what it is supposed to be done anyway?

**Comment by jcotela** [comment]
@loumalouomega remove all references to the old MKL solvers application (explicit uses of the MKL solvers).  I can do it tomorrow if you don't feel confident with the changes.

File: applications/MeshingApplication/test_examples/refine.gid/refine.prb [hunk 1]
```diff
@@ -30,7 +30,7 @@ QUESTION: Solver_Type#CB#(StaticSolver,ArcLengthSolver,LineSearchesSolver,Dynami
 VALUE: StaticSolver
```
Comment (loumalouomega) [inline applications/MeshingApplication/test_examples/refine.gid/refine.prb:30]: Thanks @jcotela

**Comment by jcotela** [comment]
@philbucher I did a couple more applications, but none of the changes had any practical impact. For the remaining applications, keep in mind that many of the occurences you detected are in `.info`, `.out` (both text output from simulations) or `.prb` (old format of GiD problemtype data) and are harmless.

@josep-m-carbonell the occurences of MKL that @philbucher detected in the PfemSolidMechanicsApplication actually correspond to a how-to in the install_scripts folder. We don't need to do anything with regards to this PR, just keep in mind that that part of the how-to will become outdated.

**Comment by philbucher** [comment]
Thx @jcotela I updated the list
I will check the applications again, but this is anyway for after the release

**Comment by philbucher** [comment]
I removed the remaining references of mkl, so this should be ready to merge

**Comment by philbucher** [comment]
@RiccardoRossi can we merge this?

File: applications/FluidDynamicsApplication/test_examples/cyl_bench/run_test.py [hunk 2]
```diff
@@ -178,9 +178,6 @@ def PrintResults(model_part):
         velocity_linear_solver = SkylineLUFactorizationSolver()
     elif ProjectParameters.Velocity_Linear_Solver == "Super LU":
         velocity_linear_solver = SuperLUSolver()
-    elif ProjectParameters.Velocity_Linear_Solver == "Parallel MKL Pardiso":
```
Comment (RiccardoRossi) [inline applications/FluidDynamicsApplication/test_examples/cyl_bench/run_test.py:181]: my suggestion would be to remove completely this test. It is old and does not match the modern testing interface. I completely agree that this should be done as a part of a different PR

File: applications/FluidDynamicsApplication/test_examples/vms2d_test/script_elemtest.py [hunk 3]
```diff
@@ -170,8 +170,6 @@ def BenchmarkCheck(time, node1, node2):
         velocity_linear_solver = SkylineLUFactorizationSolver()
     elif ProjectParameters.Velocity_Linear_Solver == "Super LU":
         velocity_linear_solver = SuperLUSolver()
-    elif ProjectParameters.Velocity_Linear_Solver == "Parallel MKL Pardiso":
```
Comment (RiccardoRossi) [inline applications/FluidDynamicsApplication/test_examples/vms2d_test/script_elemtest.py:173]: same

File: applications/applications_interface.py [hunk 4]
```diff
@@ -472,7 +462,7 @@ def ImportApplications(kernel, applications_path=application_directory):
         wind_turbine_application = KratosMultiScaleApplication()
         kernel.ImportApplication(MultiScaleApplication)
         print("KratosMultiScaleApplication sucessfully imported")
-
```
Comment (RiccardoRossi) [inline applications/applications_interface.py:475]: this file should be removed overall. again as a part of a different Pr

File: kratos/python_scripts/linear_solver_factory.py [hunk 5]
```diff
@@ -305,11 +305,6 @@ def ConstructSolver(configuration):
         scaling = params["scaling"].GetBool()
 
         linear_solver = AMGCL_NS_Solver(params)
-    #
```
Comment (RiccardoRossi) [inline kratos/python_scripts/linear_solver_factory.py:308]: can't we emulate the interface so that the corresponding interface through Eigen is used?

File: kratos/python_scripts/new_linear_solver_factory.py [hunk 6]
```diff
@@ -63,12 +63,7 @@ def ConstructSolver(configuration):
     elif(solver_type == "complex_pastix_solver"):
         import KratosMultiphysics.ExternalSolversApplication
         linear_solver = KratosMultiphysics.ExternalSolversApplication.PastixComplexSolver(configuration)
- 
-    ################################## following solvers need importing the MKLSolversApplication
-    elif (solver_type == "ParallelMKLPardisoSolver"):
```
Comment (RiccardoRossi) [inline kratos/python_scripts/new_linear_solver_factory.py:68]: let's emulate the interface through eigen. can we?

**Review by RiccardoRossi — COMMENTED** [review]
@philbucher we are very close. Only one question: could we emulate the old interface, so that if someone calls Pardiso as before, the corresponding capabilities are called using Eigen?

File: applications/applications_interface.py (hunk 4 above)
Comment (jcotela) [reply applications/applications_interface.py:475]: @RiccardoRossi I have the feeling that it is time to clean up and update our python interface, maybe as a goal for the next release, and removing this should be part of it.

File: kratos/python_scripts/linear_solver_factory.py (hunk 5 above)
Comment (philbucher) [reply kratos/python_scripts/linear_solver_factory.py:308]: sure, just a question whether I should or not.
But if you ask for it I can do it

File: kratos/python_scripts/new_linear_solver_factory.py (hunk 6 above)
Comment (philbucher) [reply kratos/python_scripts/new_linear_solver_factory.py:68]: Should be possible

File: kratos/python_scripts/linear_solver_factory.py (hunk 5 above)
Comment (RiccardoRossi) [reply kratos/python_scripts/linear_solver_factory.py:308]: I mean, what is your suggestion? - i mean, what should i do after this PR to use Pardiso?

**Comment by philbucher** [comment]
@RiccardoRossi done, now the solvers you pointed out are emulated through Eigen

File: kratos/python_scripts/new_linear_solver_factory.py (hunk 6 above)
Comment (philbucher) [reply kratos/python_scripts/new_linear_solver_factory.py:68]: done

File: kratos/python_scripts/linear_solver_factory.py (hunk 5 above)
Comment (philbucher) [reply kratos/python_scripts/linear_solver_factory.py:308]: Dont have a better suggestion, I followed your proposal

Btw if you want to use the pardiso why not using it through Eigen?

File: kratos/python_scripts/new_linear_solver_factory.py [hunk 7]
```diff
@@ -83,12 +78,17 @@ def ConstructSolver(configuration):
         import KratosMultiphysics.EigenSolversApplication
         linear_solver = KratosMultiphysics.EigenSolversApplication.PardisoLUSolver(configuration)
 
+     # emulating the solvers of the MKLSolversApplication through the EigenSolversApplication
+    elif (solver_type == "ParallelMKLPardisoSolver"):
+        import KratosMultiphysics.EigenSolversApplication
```
Comment (RiccardoRossi) [inline kratos/python_scripts/new_linear_solver_factory.py]: maybe we could even have a message saying DEPRECATED: please use KratosMultiphysics.EigenSolversApplication.PardisoLUSolver

anyway what you did is fine to me (thx)

**Review by RiccardoRossi — DISMISSED** [review]

**Comment by philbucher** [comment]
@RiccardoRossi done, pls reapprove :)

**Review by RiccardoRossi — APPROVED** [review]
